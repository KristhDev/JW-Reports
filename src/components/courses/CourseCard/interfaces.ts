import { Course } from '../../../interfaces/courses';

export interface CourseCardProps {
    course: Course;
    onActiveOrSuspend: () => void;
    onDelete: () => void;
    onFinishOrStart: () => void;
}