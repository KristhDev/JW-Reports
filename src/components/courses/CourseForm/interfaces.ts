export interface CourseFormValues {
    person_name: string;
    person_about: string;
    person_address: string;
    publication: string;
}