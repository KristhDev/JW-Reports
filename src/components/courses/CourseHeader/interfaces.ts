export interface RevisitHeaderProps {
    editButton: boolean;
    deleteButton: boolean;
}