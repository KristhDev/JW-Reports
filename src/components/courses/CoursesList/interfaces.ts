import { CourseFilter } from '../../../interfaces/courses';

export interface CoursesListProps {
    emptyMessage: string;
    filter: CourseFilter;
    title: string;
}

export interface ListEmptyComponentProps {
    msg: string;
}