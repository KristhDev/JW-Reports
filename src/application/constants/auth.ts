import { ItemOption, Precursor } from '@infrasturcture/interfaces';

/**
 * The precursors options to select
 */
export const PRECURSORS_OPTIONS: ItemOption[] = [
    { label: 'Ninguno', value: 'ninguno' },
    { label: 'Auxiliar', value: 'auxiliar' },
    { label: 'Regular', value: 'regular' },
    { label: 'Especial', value: 'especial' }
];

export const precursors: Record<Uppercase<Precursor>, Precursor> = {
    AUXILIAR: 'auxiliar',
    ESPECIAL: 'especial',
    NINGUNO: 'ninguno',
    REGULAR: 'regular'
}

export const authMessages = {
    CONFIRM_PASSWORD_EMPTY: 'La confirmación de la contraseña es requerida.',
    EMAIL_ALREADY_REGISTERED: 'Ya existe un usuario con este correo.',
    EMAIL_EMPTY: 'El correo no puede estar vacío.',
    EMAIL_INVALID: 'Correo electrónico inválido.',
    EMAIL_UPDATE_UNCHANGED: 'Para actualizar tu correo debes cambiarlo.',
    NAME_EMPTY: 'El nombre no puede estar vacío.',
    NAME_MIN_LENGTH: 'El nombre debe tener al menos 2 caracteres.',
    PASSWORD_EMPTY: 'La contraseña no puede estar vacía.',
    PASSWORD_MIN_LENGTH: 'La contraseña debe tener al menos 6 caracteres.',
    PASSWORD_NOT_MATCH: 'Las contraseñas no coinciden.',
    PASSWORD_UPDATED: 'Ha actualizado su contraseña correctamente.',
    PROFILE_UPDATED: 'Ha actualizado su perfil correctamente.',
    SURNAME_EMPTY: 'Los apellidos no pueden estar vacíos.',
    SURNAME_MIN_LENGTH: 'Los apellidos deben tener al menos 2 caracteres.',
    UNATHENTICATED: 'Para realizar está acción debe iniciar sesión.',
    UNAUTHORIZED: 'No tiene permiso para realizar está acción.',
}

export const precursorMessages = {
    PRECURSOR_EMPTY: 'El campo precursor es requerido.',
    PRECURSOR_INVALID: 'Por favor seleccione una opción de precursor.',
}