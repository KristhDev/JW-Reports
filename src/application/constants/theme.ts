import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];