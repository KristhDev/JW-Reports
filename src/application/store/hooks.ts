import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();