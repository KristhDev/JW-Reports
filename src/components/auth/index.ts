export { CredentialsForm } from './CredentialsForm';
export { ForgotPasswordForm } from './ForgotPasswordForm';
export { LoginForm } from './LoginForm';
export { ProfileForm } from './ProfileForm';
export { RegisterForm } from './RegisterForm';