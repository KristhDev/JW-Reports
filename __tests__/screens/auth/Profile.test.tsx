import React from 'react';
import { act, render, screen, waitFor } from '@testing-library/react-native';

import { Profile } from '../../../src/screens/auth';

import { useAuth, useStatus, useTheme } from '../../../src/hooks';

import { User } from '../../../src/interfaces/auth';

import { darkColors } from '../../../src/theme';

const testUser: User = {
    id: '05ef0d0c-0f7a-4512-b705-6da279d88503',
    name: 'Celestino',
    surname: 'Wilderman',
    email: 'Ernestine_Doyle@yahoo.com',
    precursor: 'ninguno',
    createdAt: '2021-03-10T12:00:00.000Z',
    updatedAt: '2021-03-10T12:00:00.000Z',
}

jest.mock('../../../src/hooks/useAuth.ts');
jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <Profile /> screen', () => {
    (useAuth as jest.Mock).mockReturnValue({
        state: { user: testUser, isAuthLoading: false },
        updateProfile: jest.fn()
    });

    (useStatus as jest.Mock).mockReturnValue({
        setErrorForm: jest.fn()
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
    });

    it('should to match snapshot', async () => {
        await waitFor(() => {
            render(<Profile />);
        });

        await act(() => {
            expect(screen.toJSON()).toMatchSnapshot();
        });
    });

    it('should render respective title', async () => {
        await waitFor(() => {
            render(<Profile />);
        });

        await act(async () => {
            await waitFor(() => {
                const title = screen.getByTestId('title-text');

                expect(title).toBeTruthy();
                expect(title.props.children).toBe('Mi perfil');
            });
        })
    });
});