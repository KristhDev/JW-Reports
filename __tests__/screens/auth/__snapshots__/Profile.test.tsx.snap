// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Profile /> screen should to match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
          "paddingTop": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 38,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 24,
          },
        ]
      }
      testID="title-text"
    >
      Mi perfil
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-field-label"
      >
        Nombre:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              cursorColor="#C0A7E1"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSelectionChange={[Function]}
              placeholder="Ingrese su nombre"
              placeholderTextColor="#BFBFBF"
              selection={
                Object {
                  "end": 9,
                  "start": 9,
                }
              }
              selectionColor="#A0B9E2"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                    "paddingRight": 5,
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
              testID="form-field-text-input"
              value="Celestino"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#BFBFBF",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-field-label"
      >
        Apellidos:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              cursorColor="#C0A7E1"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSelectionChange={[Function]}
              placeholder="Ingrese su apellido"
              placeholderTextColor="#BFBFBF"
              selection={
                Object {
                  "end": 9,
                  "start": 9,
                }
              }
              selectionColor="#A0B9E2"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                    "paddingRight": 5,
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
              testID="form-field-text-input"
              value="Wilderman"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#BFBFBF",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-select-label"
      >
        Precursor:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 0,
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderRadius": 5,
                  "flex": 1,
                }
              }
              testID="form-select-touchable"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "padding": 10,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "flex": 1,
                        "fontSize": 17,
                      },
                      undefined,
                    ]
                  }
                  testID="form-select-text-value"
                >
                  Ninguno
                </Text>
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#BFBFBF",
                        "fontSize": 25,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      statusBarTranslucent={true}
      transparent={true}
      visible={false}
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Array [
            Object {
              "backgroundColor": "#C0A7E1",
              "borderRadius": 10,
            },
            Object {
              "marginVertical": 30,
              "paddingHorizontal": 20,
            },
          ],
        ]
      }
      testID="button-touchable"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 20,
              "paddingVertical": 8,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "color": "#121212",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 2,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
          testID="button-text"
        >
          Guardar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
